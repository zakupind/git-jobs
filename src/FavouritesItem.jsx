import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import close from './img/close.svg';

import { delFavourites } from './store/action';

export function FavouritesItem(props) {
  const {
    url,
    title,
    id,
    delFavourites: delFav,
  } = props;

  return (
    <div className="favourites-item-wrapper">
      <h4 className="favourites-item__title"><a href={url}>{title}</a></h4>
      <button className="favourites-item__button_del" type="button" onClick={() => delFav(id)}>
        <img className="favourites-item__button_img_del" alt="delete" src={close} />
      </button>
    </div>
  );
}

const mapDispatchToProps = {
  delFavourites,
};

FavouritesItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  delFavourites: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(FavouritesItem);
