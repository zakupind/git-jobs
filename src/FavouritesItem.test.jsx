import enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import renderer from 'react-test-renderer';
import { FavouritesItem } from './FavouritesItem';

enzyme.configure({ adapter: new Adapter() });

describe('Тест компонента избранной вакансии', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      id: '3a591fde-815f-45e7-81e9-9781de9596a5',
      title: 'Applikasjonsutvikler med erfaring søkes til Stingray Marine Solutions',
      url: 'https://jobs.github.com/positions/3a591fde-815f-45e7-81e9-9781de9596a5',
      delFavourites: jest.fn(),
    };

    component = mount(<FavouritesItem key={props.id} {...props} />);
  });

  it('Проверка на render компонента', () => {
    expect(component).toBeTruthy();
  });

  it('Проверка пришли ли пропсы', () => {
    expect(component.props()).toEqual(props);
  });

  it('Проверка текста заголовка', () => {
    const title = component.find('.favourites-item__title');
    expect(title.text()).toEqual(props.title);
  });

  it('render FavouriteItem', () => {
    const tree = renderer
      .create(<FavouritesItem key={props.id} {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('нажатие кнопки del', () => {
    component.find('button').simulate('click');
    expect(component.props().delFavourites.mock.calls.length).toBe(1);
  });
});
