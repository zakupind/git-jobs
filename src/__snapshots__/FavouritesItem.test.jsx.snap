// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента избранной вакансии render FavouriteItem 1`] = `
<div
  className="favourites-item-wrapper"
>
  <h4
    className="favourites-item__title"
  >
    <a
      href="https://jobs.github.com/positions/3a591fde-815f-45e7-81e9-9781de9596a5"
    >
      Applikasjonsutvikler med erfaring søkes til Stingray Marine Solutions
    </a>
  </h4>
  <button
    className="favourites-item__button_del"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Убрать из избранного"
      className="favourites-item__button_img_del"
      src="close.svg"
    />
    <span>
      Удалить
    </span>
  </button>
</div>
`;
